kornia/feature/sold2/sold2.py

```diff
@@ -86,9 +87,7 @@ def __init__(self, pretrained: bool = True, config: Optional[Dict] = None):
         # Load the pre-trained model
         self.model = SOLD2Net(self.config)
         if pretrained:
-            pretrained_dict = torch.hub.load_state_dict_from_url(
-                urls["wireframe"], map_location=lambda storage, loc: storage
-            )
+            pretrained_dict = torch.hub.load_state_dict_from_url(urls["wireframe"], map_location=map_location_to_cpu)
```

Comment:
How does map location cpu work ? I’m seeing in the definition you need to pass a string ?

Reply:
From pytorch docs:

```python
>>> torch.load('tensors.pt')
# Load all tensors onto the CPU
>>> torch.load('tensors.pt', map_location=torch.device('cpu'))
# Load all tensors onto the CPU, using a function
>>> torch.load('tensors.pt', map_location=lambda storage, loc: storage)
# Load all tensors onto GPU 1
>>> torch.load('tensors.pt', map_location=lambda storage, loc: storage.cuda(1))
# Map tensors from GPU 1 to GPU 0
>>> torch.load('tensors.pt', map_location={'cuda:1':'cuda:0'})
# Load tensor from io.BytesIO object
>>> with open('tensor.pt', 'rb') as f:
...     buffer = io.BytesIO(f.read())
>>> torch.load(buffer)
# Load a module with 'ascii' encoding for unpickling
>>> torch.load('module.pt', encoding='ascii')
```

I just create the `map_location_to_cpu`, because of the typing. So, the function, is just a return of the `storage` parameter but with types. Pytorch have a [function](https://github.com/pytorch/pytorch/blob/37d54239c7ea88fd9c98dcac3fcc9b98a6f9e9d1/torch/serialization.py#L1062) to "magically" understand this.

But for older torch versions, the type annotations of `map_locations` seems to just accept a `dict[str, str]` or a `callable[[str], str]`

Reply:
sounds good -- was just curious